### PR TITLE
Preserve literal semantics in strict2 case/when

### PR DESCRIPTION
**Pull request by dejmedus**
Previously, `strict2` used `safe_parse_expression` to parse `when` expressions causing `blank` and `empty` to be treated as string literals (Expression::LITERALS maps 'empty' => ''), rather than method literals

This caused unexpected behaviour:

```liquid
{%- case products -%}
  {%- when empty -%}
    previously: doesn't render (products == '' is false)
    now: renders (products.empty? is true)
{%- endcase -%}
```

This PR instead calls `Condition.parse_expression` with `safe: true`, which will correctly handle `blank`/`empty`

<details><summary>Example</summary>

example
</details>

### CODE REVIEW

File: test/unit/tags/case_tag_unit_test.rb [hunk 1]
```diff
@@ -82,6 +82,26 @@ def test_case_when_with_or
     end
   end
 
+  def test_case_when_empty
```
Comment (dejmedus) [inline test/unit/tags/case_tag_unit_test.rb]: I'm only adding a test for `empty` as `blank` requires activesupport in order to render as we expect, however both have parity with `lax`
